pkg/kubecost/allocation.go

```diff
@@ -240,6 +245,43 @@ func (pva *PVAllocation) Equal(that *PVAllocation) bool {
 		util.IsApproximately(pva.Cost, that.Cost)
 }
 
+type ProportionalAssetResourceCost struct {
+	ProportionalAssetResourceCostKey
+	CPUPercentage float64 `json:"cpuPercentage"`
+	GPUPercentage float64 `json:"gpuPercentage"`
+	RAMPercentage float64 `json:"ramPercentage"`
+}
+
+func (parc ProportionalAssetResourceCost) Key() ProportionalAssetResourceCostKey {
+	return parc.ProportionalAssetResourceCostKey
+}
+
+type ProportionalAssetResourceCostKey struct {
+	Cluster string `json:"cluster"`
+	Node    string `json:"node"`
+}
+
+type ProportionalAssetResourceCosts map[ProportionalAssetResourceCostKey]ProportionalAssetResourceCost
+
+func (parcs ProportionalAssetResourceCosts) Insert(parc ProportionalAssetResourceCost) {
+	if curr, ok := parcs[parc.Key()]; ok {
+		parcs[parc.Key()] = ProportionalAssetResourceCost{
+			ProportionalAssetResourceCostKey: parc.Key(),
+			CPUPercentage:                    curr.CPUPercentage + parc.CPUPercentage,
```

Comment:
One note, after #1770 is landed. `CPUPercentage` and `RAMPercentage` can exceed 100%.